src/opendr/perception/object_detection_2d/utils/class_filter_wrapper.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2020-2023 OpenDR European Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from opendr.engine.target import BoundingBoxList
+from opendr.perception.object_detection_2d.centernet.centernet_learner import CenterNetDetectorLearner
+from opendr.perception.object_detection_2d.detr.detr_learner import DetrLearner
+from opendr.perception.object_detection_2d.retinaface.retinaface_learner import RetinaFaceLearner
+from opendr.perception.object_detection_2d.ssd.ssd_learner import SingleShotDetectorLearner
+from opendr.perception.object_detection_2d.yolov3.yolov3_learner import YOLOv3DetectorLearner
+from opendr.perception.object_detection_2d.yolov5.yolov5_learner import YOLOv5DetectorLearner
+from opendr.perception.object_detection_2d.nanodet.nanodet_learner import NanodetLearner
+
+
+class FilteredLearnerWrapper:
+    def __init__(self, learner, allowed_classes=None):
+        self.learner = learner
+        self.allowed_classes = allowed_classes if allowed_classes is not None else []
+
+        if isinstance(self.learner,
+                      (CenterNetDetectorLearner, YOLOv3DetectorLearner, YOLOv5DetectorLearner, NanodetLearner,
+                       RetinaFaceLearner, SingleShotDetectorLearner)):
```

Comment:
RetinaFace basically detects only faces so i am not sure whether it can be useful to filter its output. Similar to that, SSD is only a person detector.

I suggest we keep support for SSD since one may train it to detect some other class too, but drop support for RetinaFace since it only serves to detect faces. Any opinions are welcome.